docs/content/latest/develop/best-practices-ysql.md

```diff
@@ -0,0 +1,367 @@
+this does not always work well in practice because by the time hot shards are redistributed across nodes, 
+the workload could change and introduce new hot spots.
+
+#### Consistent hash as the default sharding strategy
+Defaults are meant to serve the target use case. Users turn to YugabyteDB primarily for scalability reasons. 
+Most use cases requiring scalability often do not need to perform range lookups on the primary key, hence we picked consistent 
+hash sharding as the default in YugabyteDB. User identity (user ids do not need ordering), 
+product catalog (product ids are not related to one another) and stock ticker data (one stock symbol is independent of all other stock symbol) 
+are some common examples.
+
+In cases when range queries become important after the data is already loaded into a hash sharded table, 
+a range sharded secondary index can be created on that column. Once the secondary index is rebuilt, range queries would become efficient.
+
+
+### Co-location
+Co-location is a [new feature in beta](../explore/colocated-tables/linux.md) where you can put tables of 
+a database inside 1 tablet. This increases performance and lowers latency on write transactions, joined queries and aggregations
+since they happen in 1 tablet/node.
+
+If you have a small dataset (<500GB) that requires HA, or a DB with only few scale out tables, then colocation is a good option.
+
+
+### Partial indexes
+[A partial index](../api/ysql/commands/ddl_create_index.md#where-clause) is an index that is built on a subset of a table and includes only rows that satisfy the condition 
+specified in the `WHERE` clause. It can be used to exclude `NULL` or common values from the index. 
+This will speed up any writes to the table since rows containing the common column values don't need to be indexed. 
+It will also reduce the size of the index, thereby improving the speed for read queries that use the index.
+It can also be used to index just the rows of interest. 
+For example, an application maintaining a shipments table may have a column for shipment status. 
+If the application mostly accesses the in-flight shipments, then it can use a partial index to exclude rows whose shipment status is delivered.
+
+### Use covering indexes
+When querying by a secondary index, the original table is consulted to get the columns that aren't specified in the 
+index. This can result in multiple random reads across the main table.
+
+Sometimes a better way is to include the other columns that we're querying that aren't part of the index 
+using the [`INCLUDE`](../api/ysql/commands/ddl_create_index.md#include-columns) clause.  
+When additional columns are included in the index, they can be used to respond to queries directly using index only scans
+that are faster.
+
+This turns a (possible) random read from the main table to just a filter on the index.
```

Comment:
Move this to the performance section, since this addresses how to make queries faster.